packages/light.js/src/utils/operators/distinctValues.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+//
+// SPDX-License-Identifier: MIT
+
+import BigNumber from 'bignumber.js';
+import { distinctUntilChanged } from 'rxjs/operators';
+import { isObject } from '@parity/api/lib/util/types';
+
+/**
+ * An intelligent distinctUntilChanged()
+ *
+ * @ignore
+ */
+export const distinctValues = <T>() =>
+  distinctUntilChanged<T>((x, y) => {
+    if (x instanceof BigNumber && y instanceof BigNumber) {
```

Comment:
see https://github.com/MikeMcl/bignumber.js/issues/111
related: https://github.com/parity-js/ui/pull/8#discussion_r183089995

It would be safer to use `BigNumber.isBigNumber(x) && BigNumber.isBigNumber(y)` (https://mikemcl.github.io/bignumber.js/#isBigNumber)

Reply:
Ah, right! Thanks, I'll change it asap